File: Contapoupanca.java
import java.util.Calendar;
public class Contapoupanca extends Conta{
    private Calendar data;
    private static final double TAXA = 0.2;

    public Contapoupanca (String nConta, String nomeTitular, String cpfTitular){
        super(nConta, nomeTitular, cpfTitular);
        this.data = Calendar.getInstance();
    }

    public double sacar(double valorSaque) {
        if(valorSaque < getSaldo()){
            System.out.println("Saldo indisponível");
            return getSaldo();
        }
        setSaldo(getSaldo() - valorSaque);
        return getSaldo();
    }

    public double depositar(double valorDeposito){
        setSaldo(getSaldo() + valorDeposito);
        return getSaldo();
    }

}

File: WinFormsApp1/classes/ContaPoupanca.cs


namespace WinFormsApp1
{
    public class ContaPoupanca : Conta, IEmprestimo
    {
        private Double taxaMensal;
        private Double valorPago = 0;
        public ContaPoupanca(String nConta, String senha, String nomeTitular, String cpfTitular, Double saldo)
            : base(nConta, senha, nomeTitular, cpfTitular)
        {
            this.Saldo = saldo;
            this.taxaMensal = 0.001;
        }

        public override Double sacar(Double valorSaque)
        {
            if (valorSaque > Saldo)
            {
                MessageBox.Show("Saldo Indisponivel");
                return Saldo;
            }
            Saldo = Saldo - valorSaque;
            return Saldo;
        }

        public override Double depositar(Double valorDeposito)
        {
            Saldo = Saldo + valorDeposito;
            return Saldo;
        }

        public void ObterEmprestimo(Double valor, Double taxa)
        {
            Emprestimo += valor;
            EmprestimoTaxa += taxa;
            Saldo += valor;
        }

        public void PagarEmprestimo()
        {
            double aux = Emprestimo * EmprestimoTaxa;
            if(Saldo > 0 && Emprestimo != 0 && Saldo - aux > 0)
            {
                if(valorPago < (Emprestimo + (Emprestimo * EmprestimoTaxa*10)))
                {
                    Saldo -= aux;
                    valorPago += aux;
                    Emprestimo -= aux;
                }
                else if(valorPago >= Emprestimo)
                {
                    Emprestimo = 0;
                    EmprestimoTaxa = 0;
                    valorPago = 0;
                }
            }
        }

        public void calcularJuros()
        {
            Saldo += (Saldo * taxaMensal);
        }
    }
}
